### PR TITLE
[WasmFS] Allow backends to report errors in `close`

### PR DESCRIPTION
**Pull request by tlively**
This is hard to test because we don't have a way of deterministically making a
close fail in any of our existing backends, although in principle failure can
happen in the node backend. I manually tested that this worked correctly by
synthetically injecting an error on close into the OPFS backend and checking
that it did not cause the program to crash or hang and that errno was correctly
set afterward.

However, the close method on SyncAccessHandle is infallible, so we don't
actually need any error handling. As an additional change, simplify
`wasmfs_opfs_close_access` to not wait for the file to be asynchronously closed.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #17763** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/17763" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/17763?utm_source=stack-comment).

File: src/library_wasmfs_opfs.js [hunk 1]
```diff
@@ -266,11 +266,12 @@ mergeInto(LibraryManager.library, {
...
   _wasmfs_opfs_close_access__deps: ['$wasmfsOPFSFree',
                                     '$wasmfsOPFSAccessHandles'],
-  _wasmfs_opfs_close_access: async function(ctx, accessID) {
+  _wasmfs_opfs_close_access: function(accessID) {
     let accessHandle = wasmfsOPFSAccessHandles.get(accessID);
-    await accessHandle.close();
     wasmfsOPFSFree(wasmfsOPFSAccessHandles, accessID);
-    _emscripten_proxy_finish(ctx);
+    // This is an async API, but we don't actually need to wait for it. It also
+    // cannot fail, so we don't need any kind of error handling.
```
Comment (kripken) [inline src/library_wasmfs_opfs.js:273]: Shouldn't we wait just to get synchronized in terms of timing, that is, if we wait until the close is done, then a racing open of the same location would be guaranteed to succeed?

**Review by kripken — COMMENTED** [review]
For testing, can we try something like closing a file twice? Or would that not reach this code path?

File: src/library_wasmfs_opfs.js (hunk 1 above)
Comment (tlively) [reply src/library_wasmfs_opfs.js:273]: Oh, that's a really good point. Will fix.

**Comment by tlively** [comment]
No, unfortunately (for testing purposes) if a file is closed, an invalid second close wouldn't make it into the backend.

**Review by kripken — APPROVED** [review]

**Pull request merged**